lib/index.js

```diff
@@ -3148,6 +3148,73 @@ Stream.prototype.reduce1 = function (f) {
 exposeMethod('reduce1');
 
 /**
+ * Reduces a stream, using a stream-returning function. This allows async
```

Comment:
The reason why I say that is that I guarantee you the first thing people think of when they see `flatReduce` is "async version of `reduce`", and not "monadic version of `reduce`". This is what I was thinking when I made that comment. This is partially because `flatFilter` really is the async version of `filter` (it is *not* [filterM](http://hackage.haskell.org/package/base-4.8.0.0/docs/Control-Monad.html#v:filterM)).

Results like this will confuse people, I think

```javascript
_([1, 2]).flatReduce(0, function (a, b) {
    return _([a+b, a+b]);
}); // => [3, 3, 3, 3]
```

Or we can just not call it `flatReduce`. I'm more than happy to call this `foldM`, by the way.

...or maybe I'm just being paranoid.